memo_app.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'cgi'
+require 'json'
+require 'securerandom'
+
+MEMOS_FILE = 'memos.json'
+
+not_found do
+  json(message: '404 Not Found.')
+end
+
+def load_memos
+  JSON.parse(File.read(MEMOS_FILE), symbolize_names: true)
+end
+
+def save_memos(indicated_data)
+  File.write(MEMOS_FILE, JSON.pretty_generate(indicated_data))
+end
+
+def find_memo(existing_data)
+  existing_data.find { |memo| memo[:id] == params[:id] }
+end
+
+existing_data = load_memos
+
+before do
+  @memos = existing_data
+end
+
+get '/' do
+  erb :index
+end
+
+get '/new' do
+  erb :new
+end
+
+post '/create' do
+  created_data = existing_data << { id: assign_id, title: params[:title], content: params[:content], created_at: Time.now.to_s, updated_at: Time.now.to_s }
+  save_memos(created_data)
```

Comment:
```suggestion
  existing_data << { id: assign_id, title: params[:title], content: params[:content], created_at: Time.now.to_s, updated_at: Time.now.to_s }
  save_memos(existing_data)
```

existing_data に要素を追加するなら新しい変数は用意しなくて良いと思います。